src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ClientTrackingConfiguration.kt

```diff
@@ -48,6 +56,12 @@ class ClientTrackingInterceptor : HandlerInterceptor {
     SignedJWT.parse(token.replace("Bearer ", "")).jwtClaimsSet
 
   companion object {
-    private val log = LoggerFactory.getLogger(ClientTrackingInterceptor::class.java)
+    private val log = LoggerFactory.getLogger(this::class.java)
   }
 }
+
+fun retrieveIpFromRemoteAddr(request: HttpServletRequest): String {
+  val remoteAddr = request.remoteAddr
+  val colonCount = remoteAddr.chars().filter { ch: Int -> ch == ':'.code }.count()
```

Comment:
this looks complicated - should be able to do a
```
.count { it == ':' }
```
see https://stackoverflow.com/a/63435637